benchmarks/spack/repo/packages/arepo/package.py

```diff
@@ -0,0 +1,47 @@
+import os
+
+from spack.package import *
+
+class Arepo(MakefilePackage):
+    """Arepo is a massively parallel gravity and hydrodynamics code for astrophysical simulations."""
+    homepage = "https://gitlab.mpcdf.mpg.de/vrs/arepo.git"
+    git = "https://gitlab.mpcdf.mpg.de/vrs/arepo.git"
+    
+    version("2024-04-01", commit="1fc135a0")
+   
+    depends_on('mpi')
+    depends_on('gsl')
+    depends_on('gmp')
+
+    variant('fftw', default=False, description='FFTW support')
+    variant('hdf5', default=True, description='HDF5 support')
+    variant('hwloc', default=False, description='HWLOC support')
+        
+    depends_on('fftw',  when='+fftw')
+    depends_on('hwloc', when='+hwloc')
+    
+    def edit(self, spec, prefix):
+       
+        copy('Template-Makefile.systype', 'Makefile.systype')
+        copy('Template-Config.sh', 'Config.sh')
+        
+        specific_config_path = 'examples/yee_2d/Config.sh'
+            
+        self.config_file = specific_config_path  
+
+        with open('Makefile.systype', 'w') as f:
+            f.write('SYSTYPE="Darwin"\n')
+    
```

Comment:
Is this supposed to be always set to Darwin, or only on macOS?

Reply:
I am not quite sure but here is two way to do that:

The first thing after obtaining a local copy of the code is to set a SYSTYPE variable, either by typing:

`export SYSYTPE="MySystype"`
(assuming you run the bash-shell) or by copying the Template-Makefile.systype file:

`cp Template-Makefile.systype Makefile.systype`
and uncommenting the intended SYSTYPE variable in Makefile.systype.

Reply:
I think the question here was whether always seting this to `Darwin` is correct. On my Ubuntu laptop I need to manually change it to `Ubuntu` to get the include paths right, which is not ideal. 

I would have assumed that you could just leave `SYSTYPE` undefined and spack would provide the right paths to the dependencies, but this does not seem to be the case. I didn't really dig deeper into the build system to find out what it's doing.

Reply:
This is definitely strange: I'd expect it to fail on a non-mac, which is what Tuomas is seeing, but it didn't fail for me on cosma. :thinking:

Reply:
Perhaps on cosma when you have the right modules loaded you don't need to set include paths in the makefile and the `SYSTYPE` doesn't matter. I would like the build system to work without hard-coded paths, that would make life a lot easier.

Reply:
I agree with @tkoskela, it worth to escape hard-coded paths. They could cause mess for general perspective.